code/site/components/com_pages/model/entity/page.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Joomlatools Pages
+ *
+ * @copyright   Copyright (C) 2018 Johan Janssens and Timble CVBA. (http://www.timble.net)
+ * @license     GNU GPLv3 <http://www.gnu.org/licenses/gpl.html>
+ * @link        https://github.com/joomlatools/joomlatools-framework-pages for the canonical source repository
+ */
+
+class ComPagesModelEntityPage extends KModelEntityAbstract
+{
+    public function __construct(KObjectConfig $config)
+    {
+        parent::__construct($config);
+
+        //Create the config object
+        $config = (new ComPagesObjectConfigPage())->fromFile($this->file);
+
+        //Set the properties
+        $this->setProperties($config->toArray(), false);
+
+        //Se the content
+        $this->_content = $config->getContent();
+    }
+
+    protected function _initialize(KObjectConfig $config)
+    {
+        $config->append(array(
+            'data' => array(
+                'title'       => '',
+                'summary'     => '',
+                'date'        => '',
+                'published'   => true,
+                'access'      => array(
+                    'roles'  => array('public'),
+                    'groups' => array('public', 'guest'),
+                    'redirect'    => '',
+                    'metadata'    => array(),
+                    'process'     => array(
+                        'plugins' => true
+                    ),
+                    'layout'      => 'default'
+                ),
+            )));
+
+        parent::_initialize($config);
+    }
+
+    public function content($refresh = false)
+    {
+        static $content;
+
+        if(!isset($content) || $refresh)
+        {
+            $type    = pathinfo($this->file, PATHINFO_EXTENSION);
+            $content = $this->getObject('com:pages.template')
+                ->loadString($this->_content, $type != 'html' ? $type : null, $this->path)
+                ->render();
```

Comment:
@johanjanssens,

The content here is rendered without a $data context. Thus, the `$page` entity can't be accessed from the file's content, and it can't use the properties from the frontmatter.

Reply:
Forgot about that one indeed, I have added this. I'm not injecting the page itself, but inject the frontmatter properties directly. This makes most sense since we are in page context.